src/dsql/BoolNodes.cpp

```diff
@@ -1438,14 +1393,10 @@ BoolExprNode* NotBoolNode::copy(thread_db* tdbb, NodeCopier& copier) const
 	return node;
 }
 
-bool NotBoolNode::execute(thread_db* tdbb, jrd_req* request) const
+TriState NotBoolNode::execute(thread_db* tdbb, jrd_req* request) const
 {
-	bool value = arg->execute(tdbb, request);
-
-	if (request->req_flags & req_null)
-		return false;
-
-	return !value;
+	const TriState value = arg->execute(tdbb, request);
+	return value.isUnknown() ? value : TriState(!value.value);
```

Comment:
If TriState was a real class, I'd suggest adding "bool operator!()" and simply returning !value. But now it's a typedef and so far I don't see other good reasons to change that. Sigh.